pallets/api/src/fungibles/mod.rs

```diff
@@ -214,23 +214,21 @@ pub mod pallet {
 		) -> DispatchResultWithPostInfo {
 			let who = ensure_signed(origin.clone())
 				.map_err(|e| e.with_weight(Self::weight_approve(0, 0)))?;
-			let mut current_allowance = AssetsOf::<T>::allowance(id.clone(), &who, &spender);
+			let current_allowance = AssetsOf::<T>::allowance(id.clone(), &who, &spender);
 			let spender = T::Lookup::unlookup(spender);
 			let id: AssetIdParameterOf<T> = id.into();
 
 			if value.is_zero() {
 				return Ok(Some(Self::weight_approve(0, 0)).into());
 			}
-
-			current_allowance.saturating_reduce(value);
 			// Cancel the aproval and set the new value if `current_allowance` is more than zero.
 			AssetsOf::<T>::cancel_approval(origin.clone(), id.clone(), spender.clone())
 				.map_err(|e| e.with_weight(Self::weight_approve(0, 1)))?;
-
-			if current_allowance.is_zero() {
+			let new_allowance = current_allowance.saturating_sub(value);
+			if new_allowance.is_zero() {
```

Comment:
Changed this to be more clear that it is the new allowance value that will be set.

Reply:
Thanks @Daanvdplas